Tasks/DotNetCoreInstallerV0/externals/get-os-distro.sh

```diff
@@ -156,12 +156,38 @@ get_legacy_os_name() {
     return 1
 }
 
-primaryName=$(get_current_os_name || echo "")
-legacyName=$(get_legacy_os_name || echo "")
+get_machine_architecture() {
+    eval $invocation
+
+    if command -v uname > /dev/null; then
+        CPUName=$(uname -m)
```

Comment:
Isnt this different from windows behavior - where we are relying on the process's architecture.

Reply:
.Net core installation is not not available on linux x86 currently. I have used same logic as dotnet install scripts as it ensures consistent experience for a user who has earlier tried manual installation of dotnet core.